libqtile/core/manager.py

```diff
@@ -530,19 +539,56 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        try:
+            if strut[0]:
+                x = strut[0]
+            elif strut[1]:
+                x = self.x_screen_dimensions.get('x') - strut[1]
+            else:
+                x = ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+            if strut[2]:
+                y = strut[2]
+            elif strut[3]:
+                y = self.x_screen_dimensions.get('y') - strut[2]
+            else:
+                y = ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            screen = self.find_screen(x, y)
+            assert screen is not None
+            assert screen.x is not None and screen.y is not None and \
+                   screen.height is not None and screen.width is not None
+        except AssertionError:  # strut Values are defined but the coordinates are not to be found on any screen
+            logger.error("Relevant screen could not be found")
+            return
+        except IndexError:  # if _NET_WM_STRUT_PARTIAL is not set, we need to default to something
+            logger.warning("Not enough information to calculate the relevant screen. Defaulting to current.")
+            screen = self.current_screen  # defaulting to current_screen in lack of better options?
+        except AttributeError:
+            screen = self.current_screen
+            logger.warning("A")
```

Comment:
May I suggest something like
```py
except AttributeError as e:
    logger.warning("%s. Defaulting to current screen.", e)
    screen = self.current_screen
```

Reply:
One might even go further and remove those lines again which seem to have forgotten. I thought about using this clause instead of the second assert statement. However I feel like the screen should not have None in any of those attributes so considering it just another failure to compute the screen seemed more reasonable to me.